kernel/syscall.c

```diff
@@ -3691,7 +3691,12 @@ static long _syscall(void* args_)
             /* this can return (void*)-errno */
             long ret = (long)myst_mmap(addr, length, prot, flags, fd, offset);
 
-            if (ret > 0)
+            // ATTN : always translating errors to ENOMEM is probably incorrect
+            if (ret <= 0)
```

Comment:
```suggestion
            if (ret == -1 || !ret)
```

Reply:
@bodzhang does mman return only -1 or 0 for failure?

Reply:
It can return many other negative value. The old code only translate -1/0 to ENOMEM